File: catkin_ws/src/ros_opencv/src/img_publisher.cpp
#include <ros/ros.h>
#include <image_transport/image_transport.h>
#include <cv_bridge/cv_bridge.h>
#include <opencv2/highgui/highgui.hpp>
#include <sstream>
#include <stdio.h>

int main(int argc, char** argv)
{
    ros::init(argc, argv, "img_publisher");
    ros::NodeHandle nh;
    image_transport::ImageTransport it(nh);
    image_transport::Publisher pub = it.advertise("camera/image", 1);

    cv::VideoCapture capture(0);
    cv::Mat frame;
    sensor_msgs::ImagePtr msg;

    ros::Rate loop_rate(5);
    while(nh.ok())
    {
        capture >> frame;
        if(!frame.empty())
        {
            msg = cv_bridge::CvImage(std_msgs::Header(), "bgr8", frame).toImageMsg();
            pub.publish(msg);
            printf("publishing image\n");

            cv::waitKey(1);
        }

        ros::spinOnce();
        loop_rate.sleep();
    }
}

File: catkin_ws/src/ros_opencv/src/img_publisher.py
#!/usr/bin/env python

import rospy
from sensor_msgs.msg import Image
from cv_bridge import CvBridge
import cv2

def img_talker():
    pub = rospy.Publisher('imager', Image, queue_size=10)
    rospy.init_node('img_talker', anonymous=True)
    rate = rospy.Rate(60)
    capture = cv2.VideoCapture(0)
    br = CvBridge()
    while not rospy.is_shutdown():
        ret, img = capture.read()
        rospy.loginfo('publishing image')
        pub.publish(br.cv2_to_imgmsg(img, "bgr8"))
        rate.sleep()

if __name__ == '__main__':
    try:
        img_talker()
    except rospy.ROSInterruptException:
        pass